db/blob/blob_source.cc

```diff
@@ -130,12 +133,17 @@ Status BlobSource::GetBlob(const ReadOptions& read_options,
       return Status::Corruption("Compression type mismatch when reading blob");
     }
 
+    uint64_t _bytes_read = 0;
     s = blob_file_reader.GetValue()->GetBlob(
         read_options, user_key, offset, value_size, compression_type,
-        prefetch_buffer, value, bytes_read);
+        prefetch_buffer, value, &_bytes_read);
     if (!s.ok()) {
       return s;
     }
+
+    if (bytes_read) {
+      *bytes_read = value->size();  // uncompressed blob size
```

Comment:
This is actually quite challenging. Many tests will fail once we modified this because internal stats are changing.

Reply:
How about we go the other way then and use `value_size` as `bytes_read` even if the read got served out of the cache? The values would still be consistent but we wouldn't have to update these, right? (And the bytes read/written statistics of compaction jobs would also be consistent with each other...)

Reply:
I imagine it will still fail since the read/written stats require the "blob record size" from the blob file reader, not the "value size" (only contains the blob value size)

Reply:
Yes, that's right. I'll provide an update.

Reply:
done.